tools/mqtt-bench/cmd/bench.go

```
@@ -0,0 +1,361 @@
+	quiet      bool
+	retain     bool
+	mtls       bool
+	skipTLSVer bool
+	ca         string
+)
+
+var benchCmd = &cobra.Command{
+	Use: "mqtt-bench",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) < 1 {
+			cmd.Help()
+		}
+
+		runBench()
+	},
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	benchCmd.PersistentFlags().StringVarP(&broker, "broker", "b", "tcp://localhost:1883", "address for mqtt broker, for secure use tcps and 8883")
+	benchCmd.PersistentFlags().IntVarP(&qos, "qos", "q", 0, "QoS for published messages, values 0 1 2")
+	benchCmd.PersistentFlags().IntVarP(&size, "size", "s", 100, "Size of message payload bytes")
+	benchCmd.PersistentFlags().IntVarP(&count, "count", "n", 100, "Number of messages sent per publisher")
+	benchCmd.PersistentFlags().IntVarP(&subs, "subs", "", 10, "Number of subscribers")
+	benchCmd.PersistentFlags().IntVarP(&pubs, "pubs", "", 10, "Number of publishers")
+	benchCmd.PersistentFlags().StringVarP(&format, "format", "f", "text", "Output format: text|json")
+	benchCmd.PersistentFlags().StringVarP(&conf, "config", "g", "config.toml", "config file default is config.toml")
+	benchCmd.PersistentFlags().StringVarP(&channels, "channels", "", "channels.toml", "config file for channels")
+	benchCmd.PersistentFlags().StringVarP(&ca, "ca", "", "ca.crt", "CA file")
+	benchCmd.PersistentFlags().BoolVarP(&quiet, "quiet", "", false, "Supress messages")
+	benchCmd.PersistentFlags().BoolVarP(&retain, "retain", "r", false, "Retain mqtt messages")
+	benchCmd.PersistentFlags().BoolVarP(&mtls, "mtls", "m", false, "Use mtls for connection")
+	benchCmd.PersistentFlags().BoolVarP(&skipTLSVer, "skipTLSVer", "t", false, "Skip tls verification")
+
+}
+
+func initConfig() {
+
+	if conf != "" {
+		viper.SetConfigFile(conf)
+	}
+	c := Config{}
+	viper.AutomaticEnv() // read in environment variables that match
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+
+		viper.Unmarshal(&c)
```

I think that error is ignored here. If it is, please handle it.